Developer_readme.md

```diff
@@ -0,0 +1,237 @@
+    def = "Your actions are cloaked in secrecy.",
+    off = {"Your shroud dissipates and you return to the realm of perception.", "The flash of light illuminates you - you have been discovered!"}})
+  defs_data:set('lifevision', { type = 'necromancy',
+    on = {"You narrow your eyes and blink rapidly, enhancing your vision to seek out sources of lifeforce in others.", "You already possess enhanced vision."},
+    def = "You have enhanced your vision to be able to see traces of lifeforce."})
+  defs_data:set('putrefaction', { type = 'necromancy',
+    on = {"You concentrate for a moment and your flesh begins to dissolve away, becoming slimy and wet.", "You have already melted your flesh. Why do it again?"},
+    def = "You are bathed in the glorious protection of decaying flesh.",
+    off = "You concentrate briefly and your flesh is once again solid."})
+  defs_data:set('vengeance', { type = 'necromancy',
+    staysindragon = true,
+    offline_defence = true,
+    on = {"You swear to yourself that you will wreak vengeance on your slayer.", "Vengeance already burns within your heart, Necromancer."},
+    def = "You have sworn vengeance upon those who would slay you.",
+    off = {"You forswear your previous oath for vengeance, sudden forgiveness entering your heart.", "You have sworn vengeance against none, Necromancer."}})
+end
+```
+</details>
+
+Important notes:
+- You can use defr, onr and offr whenever you need to use regex for def, defup and defoff messages respectively. They are not mutually exclusive with normal def, on, off messages so you can use normal and regex triggers on the same entry.
+- Available flags and configuration functions are:
+    - availableindragon -> boolean value that flags when a defence is available and can be put up even in dragonform (usually only for defences that also shares a general or dragon equivalent with the same name and behavior, like deathsight.
+    - invisibledef -> boolean value that determines whether the defence is invisible (not recognized as a real defence by achaea). This is useful for certain abilities that are not naturally defences per se, but still useful to keep track of.
+    - offline_defence -> boolean value that flags defences that can stay up even after logging out, see the comments in the code above for example
+    - on_enable -> custom function, this is mostly only used for monks and blademasters to assure the form they are in can use said defence. See `retaliationstrike` as an example. Can become useful in case new classes that contain different forms as well are launched.
+    - specialskip -> flags defences that for some reason you want to ignore from defup (meaning svo will skip them instead of trying to put them up before proceeding to the next defences in queue to be put up)
```

Comment:
```suggestion
    - `specialskip` -> flags defences that for some reason you want to ignore from defup in order not to get stuck (meaning svo will skip them instead of trying to put them up before proceeding to the next defences in queue to be put up)
```